zbark/debark_test.go

```diff
@@ -0,0 +1,198 @@
+package zbark
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/uber-go/zap"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/uber-common/bark"
+)
+
+func newLogrus() (bark.Logger, *bytes.Buffer) {
+	buf := &bytes.Buffer{}
+	logger := logrus.New()
+	logger.Out = buf
+	logger.Level = logrus.DebugLevel
+	logger.Formatter.(*logrus.TextFormatter).DisableTimestamp = true
+	return bark.NewLoggerFromLogrus(logger), buf
+}
+
+func newDebark() (zap.Logger, *bytes.Buffer) {
+	logrus, buf := newLogrus()
+	return Debarkify(logrus, zap.DebugLevel), buf
+}
+
+func TestLogrusOutputIsTheSame(t *testing.T) {
+	logrus, lbuf := newLogrus()
+	debark, dbuf := newDebark()
+
+	zfields := []zap.Field{
+		zap.Bool("a", true),
+		zap.Float64("b", float64(0.1)),
+		zap.Int("c", 42),
+		zap.String("d", "bar"),
+		zap.Int64("e", int64(42)),
+		zap.Stringer("g", stringable("g")),
+	}
+	const warning = "danger will robinson!"
+	logrus.WithFields(zapToBark(zfields)).Warn(warning)
+	debark.With(zfields...).Warn(warning)
+
+	require.NotZero(t, lbuf.Len(), "logrus should have logged")
+	require.NotZero(t, dbuf.Len(), "debarker should have logged")
+	assert.True(t, bytes.Equal(lbuf.Bytes(), dbuf.Bytes()), "output should be the same")
+}
+
+func TestDebark_CastNoop(t *testing.T) {
+	orig := zap.New(
+		zap.NewJSONEncoder(),
+		zap.DebugLevel,
+		zap.DiscardOutput,
+	)
+	assert.True(t, orig == Debarkify(Barkify(orig), zap.DebugLevel))
+}
+
+func TestBarkify_CastNoop(t *testing.T) {
+	b, _ := newLogrus()
+	zap := Debarkify(b, zap.DebugLevel)
+	b2 := Barkify(zap)
+
+	require.Equal(t, b, b2, "Barkify(Debarkify(bark)) should equal bark")
+	assert.True(t, b == b2, "%s", "expected barkify(debarkify(bark)) to return bark: %+v != %+v != Barkify(%+v)", b, b2, zap)
+}
+
+var levels = []zap.Level{
+	zap.DebugLevel,
+	zap.InfoLevel,
+	zap.WarnLevel,
+	zap.ErrorLevel,
+}
+
+func TestDebark_Levels(t *testing.T) {
+	logger, _ := newDebark()
+	for _, l := range append(levels, zap.PanicLevel, zap.FatalLevel) {
+		logger.SetLevel(l)
+		assert.Equal(t, l, logger.Level())
+	}
+}
+
+func TestDebark_Check(t *testing.T) {
+	logger, _ := newDebark()
+	logger.SetLevel(zap.DebugLevel)
+	for _, l := range append(levels, zap.PanicLevel, zap.FatalLevel) {
+		lc := logger.Check(l, "msg")
+		assert.NotNil(t, lc)
+		assert.True(t, lc.OK())
+	}
+
+	logger.SetLevel(zap.PanicLevel)
+	for _, l := range append(levels) {
+		assert.Nil(t, logger.Check(l, "msg"))
+	}
+
+	// We should still panic even if the level isn't enough to log.
+	logger.SetLevel(zap.FatalLevel)
+	assert.Panics(t, func() { logger.Check(zap.PanicLevel, "panic!").Write() })
+}
+
+func TestDebark_LeveledLogging(t *testing.T) {
+	logger, buf := newDebark()
+	logger.SetLevel(zap.DebugLevel)
+	for _, l := range levels {
+		buf.Reset()
+		assert.Equal(t, 0, buf.Len(), "buffer not zero")
+		logger.Log(l, "ohai")
+		assert.NotEqual(t, 0, buf.Len(), "%q did not log", l)
+		buf.Reset()
```

Comment:
do you need both the `buf.Reset()` calls? I think the one at the beginning can be removed?